src/jvmTest/kotlin/bdd/SpaceEngineersCucumberTest.kt

```diff
@@ -100,6 +109,43 @@ class SpaceEngineersCucumberTest {
         )
     }
 
+    var blockId: String? = null //973380826
+
+    private fun observeBlocks(): List<SeBlock> {
+        return environment.observe(ObservationArgs(ObservationMode.BLOCKS)).allBlocks
+    }
+
+    private fun blockToGrind(): SeBlock {
+        return observeBlocks().first { it.id == blockId }
+    }
+
+    @When("Character grinds until to {double} integrity.")
+    fun character_grinds_until_to_integrity(integrity: Double) {
+        var currentIntegrity = blockToGrind().integrity
+        environment.interact(InteractionArgs(InteractionType.EQUIP, 5, 0))
+        sleep(500)
```

Comment:
500 ms seems like unnecessarily long just for equip, have you tried less?

Reply:
I tried 50 (as per your test), but wasn't enough. 500 was second try and I left it at it.